python/do.py

```diff
@@ -0,0 +1,199 @@
+            value={
+                field_name: _gqlalchemy_type_to_mgp(ctx.graph, field_value)
+                for field_name, field_value in result.items()
+            }
+        )
+        for result in results
+    ]
+
+
+@mgp.read_proc
+def when(
+    ctx: mgp.ProcCtx,
+    condition: bool,
+    if_query: str,
+    else_query: mgp.Nullable[str] = "",
+    params: mgp.Nullable[mgp.Map] = None,
+) -> mgp.Record(value=mgp.Nullable[mgp.Map]):
+    """Depending on the value of the condition, executes `if_query` or `else_query` (read-only).
+
+    Args:
+        condition: The boolean value that determines what query to execute.
+        if_query: The read-only query to be executed if the condition is satisfied.
+        else_query: The read-only query to be executed if the condition is not satisfied.
+        params: {key: value} map of parameters applied to the given queries.
+
+    Returns:
+        value: {field_name: field_value} map containing the result records of the evaluated query.
+    """
+
+    if params is None:
+        params = {}
+
+    memgraph = gqlalchemy.Memgraph()
+
+    results = _execute_and_fetch_parametrized(
+        memgraph, if_query if condition else else_query, parameters=params
+    )
+
+    return [
+        mgp.Record(
+            value={
+                field_name: _gqlalchemy_type_to_mgp(ctx.graph, field_value)
+                for field_name, field_value in result.items()
+            }
+        )
+        for result in results
+    ]
+
+
+def _get_edge_by_id(
+    graph: mgp.Graph, edge_id: mgp.EdgeId, from_id: mgp.VertexId
+) -> mgp.Nullable[mgp.Edge]:
+    return next(
+        (
+            edge
+            for edge in graph.get_vertex_by_id(from_id).out_edges
+            if edge.id == edge_id
+        ),
+        None,
+    )
+
+
+def _gqlalchemy_type_to_mgp(graph: mgp.Graph, variable: Any) -> mgp.Any:
+    if isinstance(
+        variable,
+        (
+            bool,
+            int,
+            float,
+            str,
+            list,
+            dict,
+            datetime.date,
+            datetime.time,
+            datetime.datetime,
+            datetime.timedelta,
+        ),
+    ):
+        return variable
+
+    elif isinstance(variable, gqlalchemy.models.Node):
+        return graph.get_vertex_by_id(int(variable._id))
+
+    elif isinstance(variable, gqlalchemy.models.Relationship):
+        return _get_edge_by_id(
+            graph, edge_id=int(variable._id), from_id=int(variable._start_node_id)
+        )
+
+    elif isinstance(variable, gqlalchemy.models.Path):
+        start_id = int(variable._nodes[0]._id)
+        path = mgp.Path(graph.get_vertex_by_id(start_id))
+
+        for relationship, node in zip(variable._relationships, variable._nodes[1:]):
+            next_edge_id = int(relationship._id)
+            next_node_id = int(node._id)
+
+            edge_to_add = _get_edge_by_id(graph, edge_id=next_edge_id, from_id=start_id)
+            if edge_to_add is None:
```

Comment:
I am not really sure I understand when can this happen. If we are always exploring path from starting node and searching out edges why would we ever want to search reverse edges? Can you explain this a bit, I am sure you had a good reason for this.

Reply:
Cypher lets you match paths without taking edge direction into account ([example](https://neo4j.com/docs/cypher-manual/current/introduction/uniqueness/#_homomorphism)) even though database edges are all directed by design.
In that case, this piece of code searches for the edge in both directions.